uberfire-api/src/main/java/org/uberfire/backend/vfs/impl/ObservablePathImpl.java

```diff
@@ -246,8 +246,8 @@ void onResourceBatchEvent(@Observes final ResourceBatchChangesEvent batchEvent)
                             executeDeleteCommands();
                             break;
                         case RENAME:
-                            executeRenameCommands();
                             path = ((ResourceRenamed) change).getDestinationPath();
+                            executeRenameCommands();
```

Comment:
Hi @caponetto , Thanks for the review comments. 
When `executeRenameCommands()` method executes the base editor needs to reload and set the `VersionRecordManager` with the updated path (destination path). Earlier, `executeRenameCommands()` method used to execute first and path used to updated later. This would throw an exception in the backend when we reload the editor content with older path. This change would update the `ObservablePathImpl's` `path` field before executing the method.

Reply:
Thanks for the clarification, @akumar074!